core/types/transaction.go

```diff
@@ -64,6 +64,9 @@ type Transaction struct {
 
 	// cache of details to compute the data availability fee
 	rollupCostData atomic.Value
+
+	// optional preconditions for inclusion
+	conditional atomic.Value
```

Comment:
Need to double-check the Go version, but I believe the std-lib has Go generics for `atomic`, to not have to type-check on every Load.